### ISSUE TITLE
Provide doc-id in error message when scripting fails in a query

### ISSUE THREAD

**Issue by stu-elastic** [elastic/elasticsearch#52902]
If scripting fails in, say, an aggregation we should provide the failed document id in the error message.

This will allow them to debug their script via the [execute API](https://github.com/elastic/elasticsearch/issues/52605).

**Comment by jeffvestal** [comment]
### tldr;

If `_id` can be surfaced as part of painless stack trace, it will save a lot of time debugging painless scripts, and in turn, runtime field creation which often rely on them

### Details
I have a painless script to use in a runtime field that appeared to be working, but 1 doc out of 10,000+ was a different format that I was expecting. 
I wound up having to create a terms agg and visually scanning through all the results to look for the outlier field, then adding a script condition to handle that. 

### Example:
Using the "Detailed Listing Data" from [Airbnb Nashville](http://insideairbnb.com/get-the-data/) 202112

There is a field `bathrooms_text`. I wanted to create a new field `bathrooms` that was just the numeric value. 
Most field values were the format like `1 bath` or `2.5 baths` or `1 private bathroom`. 

When creating the field / script in the kibana runtime field UI, on first pass the script was able to dissect the field value, and convert the first item to a double. However, when I started to use the data view for things like outlier detection exceptions started being raised. 
Running the `_search` query with the script in Dev Tools, the stack trace pointed to `Double.parseDouble( tmp );` but not the doc `_id` that was causing the error. 
I eventually fell back to running a terms aggregation to find a value that did not match the parsing pattern I was expecting. I could then take that doc and use it to test my runtime field / painless script.

**Comment by jdconrad** [comment]
Closing this as it is unlikely to be prioritized.

**Comment by rjernst** [comment]
Given this is a pain point, I think we should keep this open for future prioritization.
